reporting/performance/test_fibonacci.cpp

```diff
@@ -0,0 +1,65 @@
+//  Copyright Madhur Chauhan 2020.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <benchmark/benchmark.h>
+#include <boost/math/special_functions/fibonacci.hpp>
+#include <boost/multiprecision/gmp.hpp>
+#include <utility>
+
+using T = boost::multiprecision::mpz_int;
+
+auto fib_rec(unsigned long long n) -> std::pair<T, T> {
+    if (n == 0) return {0, 1};
+    auto p = fib_rec(n >> 1);
+    T c = p.first * (2 * p.second - p.first);
+    T d = p.first * p.first + p.second * p.second;
+    return (n & 1) ? std::make_pair(d, c + d) : std::make_pair(c, d);
+}
+
+static void recursive_slow(benchmark::State &state) {
+    for (auto _ : state)
+        benchmark::DoNotOptimize(fib_rec(state.range(0)).first);
+}
+constexpr int bm_start = 1 << 3, bm_end = 1 << 22;
+BENCHMARK(recursive_slow)->Range(bm_start, bm_end);
```

Comment:
Could you add `Complexity()` here?

```cpp
BENCHMARK(recursive_slow)->Range(bm_start, bm_end)->Complexity();
```

Reply:
`Complexity(benchmark::oLogN);` results in:

```
recursive_slow_BigO          -nan lgN        -nan lgN  
recursive_slow_RMS           -nan %          -nan %    
...
iterative_fast_BigO          -nan lgN        -nan lgN  
iterative_fast_RMS           -nan %          -nan %  
```

Reply:
Oh, yeah you need to do `state.SetComplexityN(state.range(0))` after the for loop completes. Also, don't add the `benchmark::oLogN` key or else you're assuming the desired result.